### DISCUSSION TITLE
Auth.js V5 ready for production use in Nextjs applications?

### DISCUSSION THREAD

**Discussion by abhaychebium** [nextauthjs/next-auth#9511] [Help] [answered]
Is auth.js v5 ready for production use in NextJS apps? There are no widely reported known issues with using NextAuth.js version 5 in production. The package is considered production-ready and has been adopted by many users for their production applications
[1](https://www.reddit.com/r/learnjavascript/comments/13c4ag6/is_authjs_production_ready/?rdt=56765)
[2](https://authjs.dev/guides/upgrade-to-v5). While some users have reported specific issues or challenges with the library, no critical issue would preclude its use in a production environment. 

The docs still mention that they are in Beta and experimental. Is it worth switching?

**Comment by TimKieu** [comment, marked as answer]
I am also waiting for the production release v5.
So far, 2024.01.08, the V5 beta-4 still does not run smoothly on Edge runtime V8.

**Reply by abhaychebium** [reply]
you think its worth switching?

**Reply by TimKieu** [reply]
Yes, it is worth to switch whenever v5 production is ready. Then you can apply on Nextjs apps or others (Qwik) on both Vercel or Cloudflare flexibly.

**Comment by ahmad-punch** [comment]
Hey Guys, It's been some time. When should we expect this v5 to be production ready?

**Reply by alexkev** [reply]
@balazsorban44, @falcowinkler, or @ndom91 could you weigh in on this?

**Reply by ahmad-punch** [reply]
Hi folks, checking in to see if there's a timeline around this?

**Reply by max-degterev** [reply]
tbh at this point just roll your own auth. this lib is only big due to the amount of adapters. the core is small.

**Reply by HyNuwa** [reply]
any update guys?

**Reply by ragnarok22** [reply]
still nothing?

**Reply by Exp1ry** [reply]
Can we get an update? So many of us need an answer..

**Reply by VitaliyPotapov** [reply]
No answer is also an answer :(

**Reply by AliakseiPaseishviliSyntheticabio** [reply]
@balazsorban44, you are the creator of this repo, any comments on this?

**Reply by Exp1ry** [reply]
@balazsorban44 Re-tagging you for a reply.

**Reply by pricciafelio** [reply]
@balazsorban44 Re-tagging you for a reply.

**Comment by max-degterev** [comment]
@balazsorban44 @ThangHuuVu seriously though guys, we are all appreciative for your past efforts but it's clear the spark isn't there anymore. Sponsor money are nice but maybe archive the repo and put a disclaimer that it's no longer maintained. People deserve to be able to move on.

**Comment by falcowinkler** [comment]
sorry for the radio silence on this. we will get back to you soon.

**Comment by VitaliyPotapov** [comment]
From [this post:](https://www.better-auth.com/blog/authjs-joins-better-auth)

> Auth.js, formerly known as NextAuth.js, is now being maintained and overseen by Better Auth team

**Comment by max-degterev** [comment]
You mean team that committed "make me admin === true": https://www.reddit.com/r/javascript/s/5I64e6T0uz

I think with next.js 16 release we've reached the end of the road here.

**Comment by ragnarok22** [comment]
This was a great project 😔

**Comment by ndom91** [comment]
Chiming in here so that everyone who's subscribed and hasn't been following updates knows - Auth.js is being maintained by the better-auth team now and we recommend migrating there when/if time allows.

We won't be doing any new feature work at Auth.js, but any critical bugs or security issues will still be patched for a while.

**Reply by Exp1ry** [reply]
Will Authjs V5 be stepping out of beta atleast?

**Reply by Exp1ry** [reply]
@ndom91 Hi, I currently have a live platform with hundreds of users. We use AuthJS v5. Could I get a confirmation on whether AuthJS will be stepping out of beta? If not, then we might as well migrate to better-auth.

**Reply by ragnarok22** [reply]
@Exp1ry my advice: migrate to another library. I'm searching for one that accepts a backend that is not written in JS.

**Reply by max-degterev** [reply]
guys just roll your own auth. Every meme auth library out there either has downtime (clerk) or has serious security issues (better auth)

this is your wake up call to stop using vibe coded bs for critical system functions

**Reply by Exp1ry** [reply]
That's the thing, thats why AuthJS was so perfect.. To me, it was just the top layer of enterprise-grade cookie storage/management. I just used it as the Authentication layer. 

The problem with better-auth is that it forces you to intertwine your authorization logic with their pre-written system. For example, I hate how you need to use their 2FA / verification setup. With AuthJS, you pass fields into the session, and from there you can write your own logic to determine whether a user is really authorized or not (even though they are signed in already).

Sad day for us..
